### DISCUSSION TITLE
Dataflow TPU Support

### DISCUSSION THREAD

**Discussion by chrisflesher** [jax-ml/jax#17128] [Q&A] [answered]
I was wondering if there are any plans to allow TPU's in the Google Dataflow API? It looks like there is some existing support for GPU's:
https://cloud.google.com/dataflow/docs/gpu/gpu-support

I was wondering if this could help reduce our DevOps complexity vs. using `kubectl` manually.

**Comment by allenwang28** [comment]
Hey, I work at Google on Cloud TPUs. I'm not sure that there are any existing plans currently, but I am interested in learning more about your use case(s)?

**Reply by chrisflesher** [reply]
I was just wondering if it could help reduce development complexity. I was reading that Dataflow automatically manages cluster resources instead of having to do it ourselves. We are using docker / kubernetes / terraform right now but it requires a decent amount of Dev Ops support and we are a pretty small company.

**Reply by allenwang28** [reply, marked as answer]
I followed up and as of right now I am not aware of any plans for TPU integration into Dataflow. I can definitely follow up if/when that does happen though.

Out of my own curiosity - we do have other fully/semi-fully managed offerings on Google Cloud with TPU support like Vertex AI and GKE. If neither of those are viable options, I'm curious what functionality/convenience Dataflow specifically offers for you that those options do not?

**Reply by chrisflesher** [reply]
We are using GKE currently. Maybe Vertex AI is what I'm looking for, it seems to manage the infrastructure, thanks!
